### PR TITLE
chore(ci): run cppcheck-differential on modified packages

### PR DESCRIPTION
**Pull request by veqcc**
## Description

Run `cppcheck-differential` on modified packages, not on changed files.

## Related links

**Parent Issue:**

- [TIER IV internal](https://star4.slack.com/archives/C070FQMREVB/p1719877783942809)

## How was this PR tested?

Confirmed when a PR includes two or more changed packages
https://github.com/autowarefoundation/autoware.universe/actions/runs/9754082691/job/26920394028?pr=7777

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Review by kminoda — COMMENTED** [review]
7777!!!!

**Comment by veqcc** [comment]
get modified packages step is failing..
https://github.com/autowarefoundation/autoware.universe/actions/runs/9753043608/job/26917652188?pr=7777

**Comment by veqcc** [comment]
@HansRobo 
Do you have any idea how can I get full paths for modified packages?

**Comment by HansRobo** [comment]
@veqcc 

How about this?

```bash
$ colcon list --packages-select yabloc_common | awk '{print $2}' | xargs -I {} realpath {}
/home/kotaroyoshimoto/workspace/pilot-auto.xx1/src/autoware/universe/localization/yabloc/yabloc_common
```

File: .github/workflows/cppcheck-differential.yaml [hunk 1]
```diff
@@ -31,33 +38,32 @@ jobs:
...
-        if: ${{ steps.get-changed-files.outputs.changed-files != '' }}
+      - name: Run Cppcheck on modified packages
+        if: ${{ steps.get-modified-packages.outputs.modified-packages != '' }}
         continue-on-error: true
         id: cppcheck
         run: |
-          echo "Running Cppcheck on changed files: ${{ steps.get-changed-files.outputs.changed-files }}"
-          cppcheck --enable=all --inconclusive --check-level=exhaustive --error-exitcode=1 --suppressions-list=.cppcheck_suppressions --inline-suppr ${{ steps.get-changed-files.outputs.changed-files }} 2> cppcheck-report.txt
+          echo "Running Cppcheck on modified packages: ${{ steps.get-modified-packages.outputs.modified-packages }}"
+          modified-packages-with-full-path=${{ colcon list --packages-select ${{ steps.get-modified-packages.outputs.modified-packages }} | awk '{print $2}' | xargs -I {} realpath {} | tr '\n' ' ' }}
```
Comment (HansRobo) [inline .github/workflows/cppcheck-differential.yaml:51]:
```suggestion
          modified-packages-with-full-path=$( for pkg in ${{ steps.get-modified-packages.outputs.modified-packages }}; do colcon list --packages-select $pkg | awk '{print $2}' | xargs -I {} realpath {}; done
 | tr '\n' ' ' )
```

**Review by HansRobo — APPROVED** [review]

**Pull request merged**